### PR TITLE
Fix LocalThresholdTest

### PR DESCRIPTION
**Pull request by stelfrich**
This PR fixes tests to fail in order to reproduce the issues described in #474.

### CODE REVIEW

File: src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java [hunk 1]
```diff
@@ -769,4 +790,22 @@ public void testContingencyOfNormalImplementation() {
 		}
 	}
 
+	/**
+	 * Normalizes input image to [0.0, 1.0] range.
+	 */
+	private Img<DoubleType> normalizeImg(final Img<ByteType> input) {
```
Comment (dietzc) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: can't you use the `normalize` `Ops` here? if not, we should fix this?!

Comment (stelfrich) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: Good catch @dietzc! I have simplified the method in a recent commit.

The problem with `Ops.Image.Normalize` ops, however, is their restriction on the same type of input and output pixel types. Therefore, you still have to convert the input to the desired output type before applying those ops or am I missing something?

Comment (stelfrich) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: I'll investigate the bug and look into that idea afterwards (+ open an issue).

**Comment by stelfrich** [comment]
> hm, couldn't we make the `normalize` less restrictive, for example by adding a super-class which consumes an output type?

@dietzc It is possible to use `OpService.run(Ops.Image.Normalize.class, ...)` which takes care of the conversion. So this restriction only applies to the methods in the `ImageNamespace`, e.g.:

```java
public <T extends RealType<T>> IterableInterval<T> normalize(
    final IterableInterval<T> out,
    final IterableInterval<T> in) {...}
```

**Comment by dietzc** [comment]
I guess it's because we have a `Converter` from `T` to `V`. We've to check if this converter goes simply from `RealType` to `RealType` or if it only performs loss-less conversions (e.g. `Byte` to `Float` in contrast to `Float` to `Byte`).

**Comment by ctrueden** [comment]
Note that the SciJava Converter framework (i.e.: not the ImgLib2 `Converter` interface) is slated for a redesign that will explicitly differentiate between lossy and lossless conversions. This will be highly useful for ops matching, among other things.

**Comment by stelfrich** [comment]
Thanks for the input @dietzc and @ctrueden! I have opened an extra issue for the flexibility of the `ImageNamespace` (see #477), to keep track of the progress.

Do you have any major concerns regarding this PR?

File: src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java [hunk 2]
```diff
@@ -287,7 +287,7 @@ public void testLocalBernsenThreshold() {
 			new OutOfBoundsMirrorFactory<ByteType, Img<ByteType>>(Boundary.SINGLE),
 			1.0, Double.MAX_VALUE * 0.5);
 
-		assertEquals(out.firstElement().get(), true);
+		assertEquals(true, out.firstElement().get());
```
Comment (ctrueden) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: Why not use `assertTrue`?

File: src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java [hunk 3]
```diff
@@ -298,7 +298,7 @@ public void testLocalContrastThreshold() {
 		ops.run(LocalContrastThreshold.class, out, in, new RectangleShape(3, false),
 			new OutOfBoundsMirrorFactory<ByteType, Img<ByteType>>(Boundary.SINGLE));
 
-		assertEquals(out.firstElement().get(), false);
+		assertEquals(false, out.firstElement().get());
```
Comment (ctrueden) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: Why not use `assertFalse`?

**Comment by ctrueden** [comment]
No concerns on my end. I am fine with merging as-is. I did ask one question about the asserts, but it's a nitpick. Not important enough to worry about before merging.

**Comment by ctrueden** [comment]
Thanks for your efforts, @stelfrich!

File: src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java (hunk 3 above)
Comment (stelfrich) [inline src/test/java/net/imagej/ops/threshold/apply/LocalThresholdTest.java]: That's a good point! I'll address that after the merge 👍

**Pull request merged**